### PR TITLE
Update App Initialization Flow Documentation

### PR DESCRIPTION
**Pull request by gdjohnson**
This pull request reconciles our existing initialization documentation against the new flow report I wrote in mid-July.

## Link to Issue

Closes: #4762.

Paves the way for: #4582.

## Description of Changes

- Separates out Jake's old routes audit into a separate file & task
- Renames the entry from "Initialization-Flow.md" to "App-Initialization-Flow.md"
- Documents app initialization from `yarn start` to the `Sublayout` component
- Includes a number of italicized notes alerting us to codebase inconsistencies, dead code, and dead inline documentation. 
    - Some of these issues will be tackled in #4582, but if reviewers with institutional memory can comment on the notes' applicability/status, it would be tremendously helpful.

### CODE REVIEW

File: wiki/App-Initialization-Flow.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+            - _Note: The `deferChain` param, passed from Route components to the `withLayout()` function, are presently absent from `LayoutAttrs`, where they ought to be passed. If this logic is obsolete, it should be removed from our router files and `layout.tsx` documentation._
+        6. If none of these conditions apply, the routed-to page is rendered.
+        - _Note: A previous iteration of these docs, and of the inline documentation in Layout.tsx, describe conditional logic in which scopes set to Ethereum addresses should trigger a call to `initNewTokenChain`, generating a new community for the provided address while rendering a `LoadingLayout`. The relevant code is no longer present in the file, and this behavior should be considered deprecated unless explicitly reintroduced._
+9. If `selectChain()` (`/helpers/chain.ts`) is fired, per step #8:
+    1. If no `chain` argument is passed, the function defaults to a `chain` value set my `app.user.selectedChain`, or else `app.config.defaultChain`.
+        - _Note: This is likely dead code that can be  removed._
```
Comment (jnaviask) [inline wiki/App-Initialization-Flow.md:43]: Technically it's still a valid codepath -- we'd need to inspect all calls to selectChain to confirm that we can remove it.

File: wiki/App-Initialization-Flow.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        - If Network is Compound, import the Compound adapter (ETH).
+        - If Network is Aave, import the Aave adapter (ETH).
+        - If Network is ERC20 or Axie Infinity, import the (ETH) tokenAdapter.
+        - If Network is ERC721, import the (ETH) NftAdapter.
+        - If Network is SPL, import the Solana adapter.
+        - If ChainBase is Solana, import the Solana adapter.
+        - If ChainBase is Ethereum and ChainType is Offchain, import the “generic” Ethereum adapter.
+        - Otherwise, throw an “invalid chain” error.
+    5. `initServer()` is called the returned `IChainAdapter` instance, which clears local storage and makes a call to `/bulkOffchain`, whose data is used to initialize community content.
+        - _Note: Previous documentation referenced the chainEntities adapter refresh. This line is currently commented out, but lingers in the `initServer()` definition._
```
Comment (jnaviask) [inline wiki/App-Initialization-Flow.md:62]: We should remove the dead code / commented out code, and #4025 will further remove parts of the chain event loading code.

**Review by jnaviask — COMMENTED** [review]
Left some responses to notes.

File: wiki/App-Initialization-Flow.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        2. If, _at render time_, the `app.loginStatusLoaded()` method returns false, a loading spinner is shown, triggered by `shouldShowLoadingState`.
+        3. If, _at render time_, `app.config.chains.getById(scope)` does not return successfully, render a `PageNotFound` view.
+        4. If, _at render time_, there is no `selectedScope`, and no custom domain, then whatever chain is currently being loaded is deinitialized by `deinitChainOrCommunity`, and the `ScopeToLoad` state variable set to `null`. A loading spinner is shown, triggered by `shouldShowLoadingState`.
+        5. If, _at render time_, the `selectedScope` differs from `app.activeChainId()`, the `isLoading` and `scopeToLoad` global state variables are updated (to `true` and `selectedScope`, respectively). `selectChain()` is fired, receiving new scope's chain as argument; on completion, the `isLoading` state variable is set to `false`.
+            - _Note: The `deferChain` param, passed from Route components to the `withLayout()` function, are presently absent from `LayoutAttrs`, where they ought to be passed. If this logic is obsolete, it should be removed from our router files and `layout.tsx` documentation._
```
[the remark this answers is not in the document]
Comment (gdjohnson) [reply wiki/App-Initialization-Flow.md:38]: Is it worth our opening a ticket to purge `deferChain` from our CustomRoutes.tsx file?

[the remark this answers is not in the document]
Comment (jnaviask) [reply wiki/App-Initialization-Flow.md]: If it's still present there, that's a major oversight, so please open a ticket.

File: wiki/App-Initialization-Flow.md [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+7. A `Router` component within the `App` component conditionally renders if state variable `isLoading` is false; this router uses `react-router-dom` library methods to conditionally create either custom domain routes, or common domain routes, depending on whether a custom domain param is passed.
+    1. All routes are configured in the `client/scripts/navigation` directory.
+    2. `CommonDomainRoutes.tsx` and `CustomDomainRoutes.tsx` declare the routes to front-end pages, switched in `navigation/Router.tsx` based on results of the `/domain` call.
+    3. "General routes," referring to pages like `/privacy` and `/terms`, are also loaded via `GeneralRoutes.tsx`.
+8. Most of these routes are set to return their relevant views wrapped inside our `LayoutWrapper` component, via the function `withLayout()`, imported from `client/scripts/views/Layout.tsx`. 
+    1. The `LayoutWrapper` injects the page’s specific params into the page as child, and injects the `scope` and `deferChain` arguments into the `Layout` component.
```
Comment (jnaviask) [inline wiki/App-Initialization-Flow.md:30]: Should remove the deferChain reference here @gdjohnson

**Review by jnaviask — APPROVED** [review]

File: wiki/App-Initialization-Flow.md (hunk 4 above)
Comment (gdjohnson) [reply wiki/App-Initialization-Flow.md]: Good catch—done.

**Pull request merged**